upgrade/scripts/upgrade/prerequisites.sh

```diff
@@ -613,8 +614,51 @@ else
   echo "====> ${state_name} has been completed" | tee -a "${LOG_FILE}"
 fi
 
+# Pre-cache images needed for istio upgrade. As soon as cray-istio-deploy is upgraded, network
+# connection to nexus will be broken, due to istio proxy and istiod versions mismatch. Upgrade of
+# cray-istio will fix that, but images must be pre-cached for it to succeed.
+state_name="PRECACHE_ISTIO_IMAGES"
+state_recorded=$(is_state_recorded "${state_name}" "$(hostname)")
+if [[ ${state_recorded} == "0" && $(hostname) == "${PRIMARY_NODE}" ]]; then
+  echo "====> ${state_name} ..." | tee -a "${LOG_FILE}"
+  {
+    istio_images=$(yq r -j "${CSM_MANIFESTS_DIR}/platform.yaml" 'spec.charts.(name==cray-precache-images).values.cacheImages' | jq -r '.[] | select( . | contains("istio"))')
+    worker_nodes=$(grep -oP "(ncn-w\d+)" /etc/hosts | sort -u)
+    while read -r istio_image; do
+      while read -r worker_node; do
+        echo "Pre-caching image ${istio_image} on node ${worker_node}"
+        ssh -n "${worker_node}" "crictl pull ${istio_image}"
+      done <<< "${worker_nodes}"
+    done <<< "${istio_images}"
+  } >> "${LOG_FILE}" 2>&1
+  record_state "${state_name}" "$(hostname)" | tee -a "${LOG_FILE}"
+else
+  echo "====> ${state_name} has been completed" | tee -a "${LOG_FILE}"
+fi
+
+# Cleanup for Kiali configmaps
+
+state_name="KIALI CLEANUP FOR DELETING OLD CONFIGMAPS"
```

Comment:
Please avoid using spaces for state names. Something like `KIALI_CLEANUP` will suffice.